SampleApplications/SDK/Opc.Ua.Server/Server/ReverseConnectServer.cs

```diff
@@ -0,0 +1,329 @@
+    {
+        public static int DefaultReverseConnectInterval => 15000;
+        public static int DefaultReverseConnectTimeout => 30000;
+        public static int DefaultReverseConnectRejectTimeout => 60000;
+
+        public ReverseConnectServer()
+        {
+            m_connectInterval = DefaultReverseConnectInterval;
+            m_connectTimeout = DefaultReverseConnectTimeout;
+            m_rejectTimeout = DefaultReverseConnectRejectTimeout;
+            m_connections = new Dictionary<Uri, ReverseConnectProperty>();
+        }
+
+        #region StandardServer overrides
+        /// <inheritdoc />
+        protected override void OnUpdateConfiguration(ApplicationConfiguration configuration)
+        {
+            base.OnUpdateConfiguration(configuration);
+            UpdateConfiguration(configuration);
+        }
+
+
+        /// <inheritdoc />
+        protected override void StartApplication(ApplicationConfiguration configuration)
+        {
+            base.StartApplication(configuration);
+            StartTimer(false);
+        }
+
+        /// <inheritdoc />
+        protected override void OnServerStopping()
+        {
+            DisposeTimer();
+            base.OnServerStopping();
+        }
+
+        /// <inheritdoc />
+        protected override void OnServerStarting(ApplicationConfiguration configuration)
+        {
+            base.OnServerStarting(configuration);
+            UpdateConfiguration(configuration);
+            StartTimer(true);
+        }
+        #endregion
+
+        #region Public Properties
+        /// <summary>
+        /// Add a reverse connection url.
+        /// </summary>
+        public virtual void AddReverseConnection(Uri url, int timeout = 0)
+        {
+            var reverseConnection = new ReverseConnectProperty(url, timeout, false);
+            lock (m_connections)
+            {
+                m_connections[url] = reverseConnection;
+                StartTimer(false);
+            }
+        }
+
+        /// <summary>
+        /// Remove a reverse connection url.
+        /// </summary>
+        public virtual void RemoveReverseConnection(Uri url)
+        {
+            lock (m_connections)
+            {
+                m_connections.Remove(url);
+                if (m_connections.Count == 0)
+                {
+                    DisposeTimer();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Return a dictionary of configured reverse connection Urls.
+        /// </summary>
+        /// <returns></returns>
+        public virtual Dictionary<Uri, ReverseConnectProperty> GetReverseConnections()
+        {
+            lock (m_connections)
+            {
+                return m_connections;
+            }
+        }
+        #endregion
+
+        #region Private Properties
+        /// <summary>
+        /// Timer callback to establish new reverse connections.
+        /// </summary>
+        private void OnReverseConnect(object state)
```

Comment:
>object state [](start = 38, length = 12)

[Quality]   parameter state is not used, it can be removed

Reply:
its the Timercallback, can't get rid of it

---
In reply to: [432601754](https://github.com/OPCFoundation/UA-.NETStandard/pull/998#discussion_r432601754) [](ancestors = 432601754)